webpack.config.js

```diff
@@ -72,6 +75,11 @@ module.exports = {
           receivingIfStatement: '',
         }),
       }),
+      // We need to explicitly define our environment variables here so that they can be referenced in the frontend.
+      new webpack.DefinePlugin({
+        'process.env.REACT_APP_SENTRY_DSN_FRONTEND': JSON.stringify(process.env.REACT_APP_SENTRY_DSN_FRONTEND),
+        'process.env.REACT_APP_SENTRY_ENVIRONMENT': JSON.stringify(process.env.REACT_APP_SENTRY_ENVIRONMENT),
+      }),
```

Comment:
As I questioned on Slack, are there other config settings we should be passing in through this approach? For example, is this how we should be defining and passing in the API baseURL and/or the context path? Both of these values should be picked up from the config files (or environment) on the server, but just thinking out loud about whether it makes sense to define all possible config properties here. Or if we did define and default these settings here does that break the current behavior of picking up those values via environment or config files. 

Fwiw, it's hard for me to reconcile all of the possible stages and environment for which our config can and should be set.
* Build (Bamboo build that generates WAR file) -> set defaults, disable everything
* Deployment (Bamboo build task that deploys WAR file) -> nothing can be done at this point AFAIK
* Runtime (Tomcat setenv.sh / environment variable) -> override default
* Runtime (Tomcat setenv.sh -D system properties) -> override default
* Runtime (openboxes.yml, etc) -> override default (should eventually be refreshed at runtime)

The last three are managed by Ansible. The first two are managed by what's in the source code and any Bamboo build tasks.

Reply:
My idea for the future is to have bamboo build and deploy do nothing except call ansible scripts and to consolidate as much as we can into configuring things via openboxes.yml instead of environment variables.

On the backend I want to have a per-environment yml file (openboxes-obdev3.yml) that lives in the dev-ops project (or better yet a private repo that it can pull from). We can shove all environment-specific settings in those files, then the configure playbook will simply copy that into the host for obdev3. Then eventually we have bamboo running the configure playbook on every deploy. So all we'd need to do to update a setting for an environment is change it in openboxes-xyz.yml then trigger a redeploy and boom it's there.

For the frontend I'm less certain about how that would work. One possible idea is that we do something similar and create .env.obdev3 files containing env variables that are specific to each environment, then again have ansible copy that into the host as a part of the configure playbook.

https://www.opcito.com/blogs/managing-multiple-environment-configurations-in-react-app

Essentially I want to get to a place where we can merge new environment config to a file somewhere, then simply redeploy and it's there.